Skyra.Core/Cache/Models/Prompts/CorePromptStateConverter.cs

```diff
@@ -8,49 +7,32 @@ namespace Skyra.Core.Cache.Models.Prompts
 {
 	internal sealed class CorePromptStateConverter : JsonConverter
 	{
-		private Dictionary<CorePromptStateType, Func<JToken, ICorePromptState>> TypeResolvers { get; } =
-			new Dictionary<CorePromptStateType, Func<JToken, ICorePromptState>>(new[]
-			{
-				new KeyValuePair<CorePromptStateType, Func<JToken, ICorePromptState>>(
-					CorePromptStateType.MessageSingleUser,
-					ParseMessageSingleUser),
-				new KeyValuePair<CorePromptStateType, Func<JToken, ICorePromptState>>(
-					CorePromptStateType.ReactionSingleUser,
-					ParseReactionSingleUser)
-			});
-
 		public override void WriteJson(JsonWriter writer, object? value, JsonSerializer serializer)
 		{
 			throw new NotImplementedException();
 		}
 
 		[NotNull]
 		public override object? ReadJson(JsonReader reader, Type objectType, object? existingValue,
-			JsonSerializer serializer)
+			[NotNull] JsonSerializer serializer)
 		{
 			var jo = JObject.Load(reader);
 			var type = Enum.Parse<CorePromptStateType>((string) jo["type"]!);
-			var state = TypeResolvers[type](jo["s"]!);
+			var data = jo["s"]!.CreateReader();
+			ICorePromptState state = type switch
+			{
+				CorePromptStateType.MessageSingleUser => serializer.Deserialize<CorePromptStateMessage>(data)!,
+				CorePromptStateType.ReactionSingleUser => serializer.Deserialize<CorePromptStateReaction>(data)!,
+				CorePromptStateType.RichDisplay => serializer.Deserialize<CoreRichDisplay>(data)!,
+				_ => throw new ArgumentOutOfRangeException()
```

Comment:
Consider adding a message to the exception - `nameof(state)` is the most common thing to do here.